### PR TITLE
Fix #70: Perform second parse if needed to be spec compliant

### LINKED ISSUE #70: Backreferences are sometimes parsed as octal escapes
Originally reported by @nhahtdh, here: https://github.com/mathiasbynens/regexpu/issues/19

```bash
$ regjsparser '(\1)+\1\1'
{
  "type": "alternative",
  "body": [
    {
      "type": "quantifier",
      "min": 1,
      "max": null,
      "greedy": true,
      "body": [
        {
          "type": "group",
          "behavior": "normal",
          "body": [
            {
              "type": "value",
              "kind": "octal", // ←
              "codePoint": 1,
              "range": [
                1,
                3
              ],
              "raw": "\\1"
            }
          ],
          "range": [
            0,
            4
          ],
          "raw": "(\\1)"
        }
      ],
      "range": [
        0,
        5
      ],
      "raw": "(\\1)+"
    },
    {
      "type": "reference",
      "matchIndex": 1,
      "range": [
        5,
        7
      ],
      "raw": "\\1"
    },
    {
      "type": "reference",
      "matchIndex": 1,
      "range": [
        7,
        9
      ],
      "raw": "\\1"
    }
  ],
  "range": [
    0,
    9
  ],
  "raw": "(\\1)+\\1\\1"
}
```

### PR DESCRIPTION
**Pull request by jviereck**
\r? @mathiasbynens, @d10

### CODE REVIEW

File: parser.js [hunk 1]
```diff
@@ -637,6 +636,10 @@
           // then the slash is ignored and the number is matched later
           // as normal characters.
 
+          // Recall the negative decision to decide if the input must be parsed
+          // a second time with the total normal-groups.
+          backrefDenied.push(refIdx);
```
Comment (jviereck) [inline parser.js]: If anyone comes up with a better name than `backrefDenied` I would be more than welcome for it :)

**Comment by jviereck** [comment]
*PING* - @mathiasbynens, @d10, can anyone of you do a review on this change set? Any help is highly appreciated. Let me know if I can support you with the review in any ways :)

**Comment by d10** [comment]
LGTM.

File: parser.js [hunk 2]
```diff
@@ -905,6 +914,23 @@
       throw SyntaxError('Could not parse entire input - got stuck: ' + str);
     }
 
+    // The spec requires to interpret the `\2` in `/\2()()/` as backreference.
+    // As the parser collects the number of capture groups as the string is
+    // parsed it is impossible to making these decisions at the point the `\2`
```
Comment (mathiasbynens) [inline parser.js]: typo: to make

File: parser.js [hunk 3]
```diff
@@ -905,6 +914,23 @@
...
+    // handled. In case the local decision turns out to be wrong later after the
```
Comment (mathiasbynens) [inline parser.js]: typo: is handled

Comment (mathiasbynens) [inline parser.js]: -later

**Comment by mathiasbynens** [comment]
LGTM with minor nits addressed.

**Comment by jviereck** [comment]
Thanks for the review guys :) Will bump the version to `0.1.4` and do a new npm publish.

**Pull request merged**